test/FractionalGovernor.sol

```diff
@@ -7,6 +7,20 @@ import "@openzeppelin/contracts/governance/extensions/GovernorVotes.sol";
 contract FractionalGovernor is GovernorVotes, GovernorCountingFractional {
   constructor(string memory name_, IVotes token_) Governor(name_) GovernorVotes(token_) {}
 
+  function castVoteWithReasonAndParamsBySig(
+    uint256 proposalId,
+    uint8 support,
+    string calldata reason,
+    bytes memory params,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) public virtual override(GovernorCountingFractional, Governor) returns (uint256) {
+    return GovernorCountingFractional.castVoteWithReasonAndParamsBySig(
```

Comment:
Why is this method/override required? I'd have thought that by default, since we inherit from `GovernorCountingFractional` which inherits from `Governor`, calling `castVoteWithReasonAndParamsBySig` would use the `GovernorCountingFractional` version

Reply:
We also inherit from `GovernorVotes` which has its own implementation because it inherits it from `Governor`, so the compiler has no way of knowing which implementation we intend here.

Reply:
hm, feels like a weird architecture that we have two base contracts that each inherit from governor, why is that?
- `GovernorCountingFractional is Governor`
- `GovernorVotes is Governor`
- `FractionalGovernor is GovernorVotes, GovernorCountingFractional`

Reply:
That's how OZ is architected. You select several extensions that implement their portion of the logic, but all inherit from `Governor`. I don't remember it off the top of my head, but iirc there is a name for this as a OOP pattern.

Reply:
got it, that does make sense for an arch where users need to be choose multiple extensions, will mark this resolved

Reply:
> That's how OZ is architected. You select several extensions that implement their portion of the logic, but all inherit from `Governor`. I don't remember it off the top of my head, but iirc there is a name for this as a OOP pattern.

We found this pattern to be pretty painful in Frax Governance where we use 6 extensions 😄

Reply:
Good question. Tbh I'm not sure why they did it that way. Maybe someone from OZ can weigh in here

But if you look at their mocks you can see that this kind of duplication is intended, e.g.:
* [GovernorVoteMock](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/30256fa8384cd8e95de7413454bb74c913ed9f3f/contracts/mocks/governance/GovernorVoteMock.sol#L8)
* [GovernorWithParamsMock](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/30256fa8384cd8e95de7413454bb74c913ed9f3f/contracts/mocks/governance/GovernorWithParamsMock.sol#L8)

We're just following their pattern